src/notification/notification.service.ts

```diff
@@ -0,0 +1,159 @@
+
+  async getNotifications(
+    userId: number,
+    pageQuery: PageQuery,
+  ): Promise<PageResponse<NotificationInfoResponseDto>> {
+    const [records, total] = await this.notificationsRepository.findAndCount({
+      where: {
+        userId: userId,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+      skip: (pageQuery.page - 1) * pageQuery.pageSize,
+      take: pageQuery.pageSize,
+    });
+    const dtos = records.map(
+      (record) => new NotificationInfoResponseDto(record),
+    );
+    return new PageResponse<NotificationInfoResponseDto>(
+      dtos,
+      total,
+      pageQuery.page,
+      pageQuery.pageSize,
+    );
+  }
+  async getNewNotifications(
+    userId: number,
+    pageQuery: PageQuery,
+  ): Promise<PageResponse<NotificationInfoResponseDto>> {
+    const [records, total] = await this.notificationsRepository.findAndCount({
+      where: {
+        userId: userId,
+        isRead: false,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+      skip: (pageQuery.page - 1) * pageQuery.pageSize,
+      take: pageQuery.pageSize,
+    });
+    const dtos = records.map(
+      (record) => new NotificationInfoResponseDto(record),
+    );
+
+    this.notificationsRepository.save(
+      records.map((record) => ({ isRead: true, ...record })),
+    );
+
+    return new PageResponse<NotificationInfoResponseDto>(
+      dtos,
+      total,
+      pageQuery.page,
+      pageQuery.pageSize,
+    );
+  }
+
+  async sendNotification(
+    userIds: number[],
+    title: string,
+    body: string,
+  ): Promise<void> {
+    const tokens = await this.notificationTokenRepository.find({
+      where: { userId: In(userIds), is_active: true },
+    });
+    const tokenList = tokens.map((token) => token.notification_token);
+
+    const responses = await firebase.messaging().sendEachForMulticast({
+      notification: {
+        title,
+        body,
+      },
+      tokens: tokenList,
+    });
+    if (responses.failureCount > 0) {
+      Promise.all(
+        responses.responses.map((response, index) => {
+          if (response.success) return;
+          if (response.error.code === 'messaging/invalid-registration-token') {
+            return this.deleteToken(userIds[index], tokenList[index]);
+          } else if (
+            response.error.code ===
+            'messaging/registration-token-not-registered'
+          ) {
+            return this.deleteToken(userIds[index], tokenList[index]);
+          } else if ((response.error.code = 'messaging/server-unavailable')) {
+            try {
+              return firebase.messaging().send({
+                notification: {
+                  title,
+                  body,
+                },
+                token: tokenList[index],
+              });
+            } catch (err) {
+              throw new InternalServerErrorException(err);
+            }
+          } else {
+            throw new InternalServerErrorException(response.error);
+          }
```

Comment:
```suggestion
          if (response.error.code === 'messaging/invalid-registration-token') {
            return this.deleteToken(userIds[index], tokenList[index]);
          }
          if (response.error.code === 'messaging/registration-token-not-registered') {
            return this.deleteToken(userIds[index], tokenList[index]);
          }
          if (response.error.code === 'messaging/server-unavailable') {
            try {
              return firebase.messaging().send({
                notification: {
                  title,
                  body,
                },
                token: tokenList[index],
              });
            } catch (err) {
              throw new InternalServerErrorException(err);
            }
          }
          throw new InternalServerErrorException(response.error);
```
이거 반영안하더라도 2번째 else if가 assignment로 되어있는건 꼭 수정해주세요

Reply:
아.